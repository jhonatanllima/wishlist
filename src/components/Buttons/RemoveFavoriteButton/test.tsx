import { screen } from '@testing-library/react';
import { renderWithTheme } from '~/utils/tests/helpers';

import { RemoveFavoriteButton } from '.';

const clickButton = jest.fn();

describe('< RemoveFavoriteButton /> ', () => {
  it('should render the correctly', () => {
    const { container } = renderWithTheme(
      <RemoveFavoriteButton onClick={() => clickButton} />
    );

    expect(screen.getByRole('button')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
