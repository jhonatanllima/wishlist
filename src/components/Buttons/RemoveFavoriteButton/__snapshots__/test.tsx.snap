// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< RemoveFavoriteButton />  should render the correctly 1`] = `
.c0 {
  width: 3rem;
  height: 3rem;
  border: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: transparent;
}

.c0:hover {
  cursor: pointer;
}

<div>
  <button
    aria-label="Remover favorito"
    class="c0"
  >
    <svg
      color="#000000"
      fill="currentColor"
      height="25"
      stroke="currentColor"
      stroke-width="0"
      style="color: rgb(0, 0, 0);"
      viewBox="0 0 1024 1024"
      width="25"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M685.4 354.8c0-4.4-3.6-8-8-8l-66 .3L512 465.6l-99.3-118.4-66.1-.3c-4.4 0-8 3.5-8 8 0 1.9.7 3.7 1.9 5.2l130.1 155L340.5 670a8.32 8.32 0 0 0-1.9 5.2c0 4.4 3.6 8 8 8l66.1-.3L512 564.4l99.3 118.4 66 .3c4.4 0 8-3.5 8-8 0-1.9-.7-3.7-1.9-5.2L553.5 515l130.1-155c1.2-1.4 1.8-3.3 1.8-5.2z"
      />
      <path
        d="M512 65C264.6 65 64 265.6 64 513s200.6 448 448 448 448-200.6 448-448S759.4 65 512 65zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
      />
    </svg>
  </button>
</div>
`;
